src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -775,6 +776,16 @@ public void terminateTask(String taskId, TerminateInfo terminateInfo) {
         }
     }
 
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public TaskResource updateTaskIndex(String taskId) {
+        TaskResource taskResource = cftTaskDatabaseService.findByIdOnly(taskId).orElse(null);
+        if (taskResource != null) {
+            taskResource.setIndexed(true);
+            cftTaskDatabaseService.saveTask(taskResource);
+        }
+        return taskResource;
```

Comment:
Is there a potential race condition here?
If you have 2 threads both reading the same data with slightly different updates then what would happen?
Postgres has a default transaction isolation level of read_committed and we are saying here that these would be separate transactions.  So whoever wins the race would update and the 2nd thread will update it again in another transaction.

Reply:
changed the findTaskById method to findByIdAndObtainPessimisticWriteLock where it adds a lock before update the indexed flag. I am going to add a timeout to aquire a lock In RWA-2216 as I have done some changes in updateTaskIndex method there.